report.md

```diff
@@ -2,6 +2,24 @@
 
 ## Subsystem 1 (MFA)
 
+Multifactor authentication will apply to all users and will aim for a balance between confidentiality and availability.
+
+### First Factor
+The first step of authentication will involve a username and password. The username will be stored as a hash. The password will be stored as a hash with salt. The salt for each user will be unique and will be generated upon registration. The salt will be stored as plain text.
+
+### Second Factor
+The second factor will be triggered on the completion of the first factor. On passing the first stage a random string will be generated by the server and stored temporarily in the database with the associated user in hash form. An sms will be sent to the user based on the registered phone number with the digits. The user will be redirected to another form to key in the digits. He will have 2 minutes to do so after which the login process will fail and the temporary string will be replaced with nonces. On success the user will receive a session and be able to login and the string will be replaced with nonces.
+
+### Third factor for special actions
+The third factor will be applicable only to administrators. The administrators will have a specially assigned BLE token with a private symmetric??? key. When logging in as an administrator, the token will have to be present and scanned before gaining access to the account on top of the first and second factor.
+
+### Services and tools to be used
+GraphQL and PHP for the initial registration. PHP crypto libraries will be used to encrypt the usernames,passwords and other data.
+
+Free SMS service will be used to send the random string for the second factor [Sample free sms service](https://www.textlocal.com/integrations/api/).
+
+SSL certificates will be generate for transferring data between client and server

```

Comment:
are these SSL certificated used to ensure confidentiality? how to generate them?

Reply:
openssl to generate. theres guides online haha. We probably can get a free signed cert also but this one gotta check